appinventor/components/src/com/google/appinventor/components/runtime/util/WebRTCNativeMgr.java

```diff
@@ -230,92 +232,92 @@ public void initiate(ReplForm form, Context context, String code) {
-              Log.d(LOG_TAG, "keepPolling is false, we're done!");
-              return;
-            }
-            Log.d(LOG_TAG, "response = " + sb.toString());
-            JSONArray jsonArray = new JSONArray(sb.toString());
-            Log.d(LOG_TAG, "jsonArray.length() = " + jsonArray.length());
-            int i = 0;
-            while (i < jsonArray.length()) {
-              Log.d(LOG_TAG, "i = " + i);
-              Log.d(LOG_TAG, "element = " + jsonArray.optString(i));
-              JSONObject element = (JSONObject) jsonArray.get(i);
-              if (!haveOffer) {
-                if (!element.has("offer")) {
-                  i++;
-                  continue;
-                }
-                JSONObject offer = (JSONObject) element.get("offer");
-                String sdp = offer.optString("sdp");
-                String type = offer.optString("type");
-                Log.d(LOG_TAG, "sdb = " + sdp);
-                Log.d(LOG_TAG, "type = " + type);
-                haveOffer = true;
-                Log.d(LOG_TAG, "About to set remote offer");
-                peerConnection.setRemoteDescription(sdpObserver,
-                  new SessionDescription(SessionDescription.Type.OFFER, sdp));
-                peerConnection.createAnswer(sdpObserver, new MediaConstraints());
-                Log.d(LOG_TAG, "createAnswer returned");
-                i = -1;
-              } else {
-                if (element.has("nonce")) {
-                  String nonce = element.optString("nonce");
-                  if (element.isNull("candidate")) {
-                    Log.d(LOG_TAG, "Received a null candidate, skipping...");
-                    i++;
-                    continue;
-                  }
-                  JSONObject candidate = (JSONObject) element.get("candidate");
-                  String sdpcandidate = candidate.optString("candidate");
-                  String sdpMid = candidate.optString("sdpMid");
-                  int sdpMLineIndex = candidate.optInt("sdpMLineIndex");
-                  Log.d(LOG_TAG, "candidate = " + sdpcandidate);
-                  if (!seenNonces.contains(nonce)) {
-                    seenNonces.add(nonce);
-                    Log.d(LOG_TAG, "new nonce, about to add candidate!");
-                    IceCandidate iceCandidate = new IceCandidate(sdpMid, sdpMLineIndex, sdpcandidate);
-                    peerConnection.addIceCandidate(iceCandidate);
-                    Log.d(LOG_TAG, "addIceCandidate returned");
-                  }
-                }
-              }
+    try {
+      if (!keepPolling) {
+        return;
+      }
+
+      Log.d(LOG_TAG, "Poller() Called");
+      HttpClient client = new DefaultHttpClient();
+      HttpGet request = new HttpGet("http://" + rendezvousServer + "/rendezvous2/" + rCode + "-s");
+      HttpResponse response = client.execute(request);
+      StringBuilder sb = new StringBuilder();
+
+      BufferedReader rd = new BufferedReader
```

Comment:
Shame on me for not catching this in the WebRTC review, but `rd` is never closed. Add a `finally` block and close it.